cypress/e2e/shared/secrets.test.ts

```diff
@@ -0,0 +1,82 @@
+import {Organization} from '../../src/types'
+
+describe('Secrets', () => {
+  beforeEach(() => {
+    cy.flush()
+    cy.signin().then(() => {
+      cy.get('@org').then(({id}: Organization) =>
+        cy.fixture('routes').then(({orgs}) => {
+          cy.visit(`${orgs}/${id}/settings/`)
+          cy.getByTestID('tree-nav')
+          cy.setFeatureFlags({secretsUI: true}).then(() => {
+            cy.getByTestID('secrets--tab').click()
+          })
+        })
+      )
+    })
+  })
+
+  it('Secrets tab setup', () => {
+    // Empty state exists and displays context appropriate text
+    cy.getByTestID('empty-state').should('be.visible')
+    cy.getByTestID('empty-state').contains('Secrets')
+
+    // Create single secret, make sure it's visible
+    cy.get('@org').then(({id: orgID}: Organization) => {
+      cy.upsertSecret(orgID, {toEverybody: 'rupees baby'})
+        .then(() => {
+          cy.reload()
+          cy.getByTestID('secret-card--toEverybody').should('be.visible')
+          cy.getByTestID('copy-to-clipboard--toEverybody').should('exist')
+
+          // Test copy to clipboard via button
+          cy.getByTestID('copy-to-clipboard--toEverybody')
+            .click({force: true})
+            .then(() => {
+              cy.task('getClipboard').should('eq', 'toEverybody')
+            })
+        })
+        // Create a second secret, make sure it's visible and test sorting
+        .upsertSecret(orgID, {CocaColaRecipe: 'lol'})
+        .then(() => {
+          cy.reload()
+          cy.getByTestID('secret-card--toEverybody').should('exist')
```

Comment:
can we validate that the original one is still there?

Reply:
This one is the initial one - I checked existence on both rather than visibility.  I can add a check that both are visible as well if you'd like?

Reply:
Sorry, yup, that's what I meant. Thanks!

Reply:
Added